utils/configuration.go

```diff
@@ -222,3 +222,53 @@ func ParseViper(v *viper.Viper, configFile string) error {
 	}
 	return nil
 }
+
+// AdjustLogLevel increases/decreases the log level, return error if the operation is invaild.
+func AdjustLogLevel(increment bool) error {
+	lvl := logrus.GetLevel()
+
+	// The log level seems not possible, in the foreseeable future,
+	// out of range [Panic, Debug]
+	if increment {
+		if lvl == logrus.DebugLevel {
+			return fmt.Errorf("log level can not be set higher than %s", "Debug")
+		}
+		lvl++
+	} else {
+		if lvl == logrus.PanicLevel {
+			return fmt.Errorf("log level can not be set lower than %s", "Panic")
+		}
+		lvl--
+	}
+
+	logrus.SetLevel(lvl)
+	return nil
+}
+
+// ReloadConfiguration reads and parse the configuration in the host and do some verification.
+func ReloadConfiguration(oldConfig *viper.Viper, configFile string, envPrefix string, reload func(*viper.Viper)) error {
+	logrus.Infof("Got signal to reload configuration, reloading from: %s", configFile)
+
+	// parse a viper as the new Config
+	newConfig := viper.New()
+	SetupViper(newConfig, envPrefix)
+	if err := ParseViper(newConfig, configFile); err != nil {
+		return err
+	}
+
+	// To check if there was anything wrong
+	if err := validateConfiguration(oldConfig, newConfig); err != nil {
+		return err
+	}
+
+	reload(newConfig)
+	return nil
+}
+
+// Reserved for future use, we can use it to validate some specific configs when necessary.
+// For example:
+// - to check if there are conflicts.
+// - to do some sanity check maybe.
+func validateConfiguration(oldConfig, newConfig *viper.Viper) error {
```

Comment:
could we do some checking for log level here for now?  If the log-level is valid at least?  It also could be helpful to log if the config is the same or different if we're only sending signals.